distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection.enso

```diff
@@ -107,27 +107,47 @@ type Connection
             if all_fields then renamed else
                 renamed.select_columns ["Database", "Schema", "Name", "Type", "Description"]
 
-    ## Set up a query returning a Table object, which can be used to work with data within the database or load it into memory.
+    ## Set up a query returning a Table object, which can be used to work with
+       data within the database or load it into memory.
 
        Arguments:
        - query: name of the table or sql statement to query.
-         If supplied as `Text`, the name is checked against the `tables` list to determine if it is a table or a query.
+         If supplied as `Text`, the name is checked against the `tables` list to
+         determine if it is a table or a query.
        - alias: optionally specify a friendly alias for the query.
-    query : Text | SQL_Query -> Text -> Table
-    query self query alias="" = handle_sql_errors <| case query of
+
+       ! Error Conditions
+
+         - If provided with a `Raw_SQL` query or a text short-hand that looks
+           like a query (currently the heuristic is to check if the name
+           contains `SELECT`), if any SQL error occurs when executing the query,
+           an `SQL_Error` dataflow error is raised.Error
+         - If provided with a `Table_Name` or a text short-hand and the table is
+           not found, a `Table_Not_Found` dataflow error is raised.
+    query : Text | SQL_Query -> Text -> Table ! Table_Not_Found | SQL_Error
+    query self query alias="" = case query of
         _ : Text ->
-            self.query alias=alias <|
+            result = self.query alias=alias <|
                 if self.tables.at 'Name' . to_vector . contains query then (SQL_Query.Table_Name query) else
                     SQL_Query.Raw_SQL query
-        SQL_Query.Raw_SQL raw_sql ->
+            result.catch SQL_Error sql_error->
+                is_probably_a_query = query.contains "SELECT"
```

Comment:
This should probably be a method on Connection.
We should allow each type to determine it.
`EXEC` probably worth checking for as well (stored procedure).